opentech/public/projects/models.py

```diff
@@ -0,0 +1,152 @@
+from django.db import models
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.paginator import EmptyPage, PageNotAnInteger, Paginator
+from django.core.validators import URLValidator
+
+from modelcluster.fields import ParentalKey
+from wagtail.wagtailadmin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    PageChooserPanel,
+    StreamFieldPanel,
+)
+
+from wagtail.wagtailcore.fields import StreamField
+from wagtail.wagtailimages.edit_handlers import ImageChooserPanel
+from wagtail.wagtailsearch import index
+
+from opentech.public.utils.blocks import StoryBlock
+from opentech.public.utils.models import (
+    BaseFunding,
+    BasePage,
+    FundingMixin,
+    RelatedPage,
+)
+
+
+class ProjectContactDetails(models.Model):
+    project_page = ParentalKey(
+        'ProjectPage',
+        related_name='contact_details'
+    )
+    site_titles = (
+        ('website', "Main Website URL"),
+        ('twitter', "Twitter Handle"),
+        ('github', "Github Organisation or Project"),
+    )
+    site_urls = (
+        ('website', ''),
+        ('twitter', 'https://twitter.com/'),
+        ('github', 'https://github.com/'),
+    )
+    service = models.CharField(
+        max_length=200,
+        choices=site_titles,
+    )
+    value = models.CharField(max_length=255)
+
+    @property
+    def url(self):
+        return dict(self.site_urls)[self.service] + self.value
+
+    def service_name(self):
+        site_display = {
+            'twitter': '@' + self.value,
+            'github': 'Github',
+            'website': 'Main Website',
+        }
+        return site_display[self.service]
+
+    def clean(self):
+        if self.service == 'twitter' and self.value.startswith('@'):
+            self.username = self.username[1:]
+
+        if self.service == 'website':
+            validate = URLValidator()
+            try:
+                validate(self.value)
+            except ValidationError as e:
+                raise ValidationError({'value': e})
+
+
+class ProjectPageRelatedPage(RelatedPage):
+    source_page = ParentalKey('ProjectPage', related_name='related_pages')
+
+    panels = [
+        PageChooserPanel('page', 'projects.ProjectPage'),
+    ]
+
+
+class ProjectFunding(BaseFunding):
+    page = ParentalKey('ProjectPage', related_name='funding')
+
+
+class ProjectPage(FundingMixin, BasePage):
```

Comment:
This is missing the categories:
Core issues (select)
Objectives (select)
Beneficiaries (select)
Addressed problems (select)
Technology focus (select)
Regions (select)
Countries (select)